localstack/services/apigateway/provider.py

```diff
@@ -59,14 +62,37 @@
 from localstack.services.edge import ROUTER
 from localstack.services.moto import call_moto
 from localstack.services.plugins import ServiceLifecycleHook
-from localstack.utils.collections import PaginatedList, ensure_list, select_from_typed_dict
+from localstack.utils.collections import (
+    DelSafeDict,
+    PaginatedList,
+    ensure_list,
+    select_from_typed_dict,
+)
 from localstack.utils.json import parse_json_or_yaml
 from localstack.utils.strings import short_uid, str_to_bool, to_str
 from localstack.utils.time import now_utc
 
 LOG = logging.getLogger(__name__)
 
 
+def get_moto_backend(context: RequestContext) -> apigw_models.APIGatewayBackend:
+    return apigw_models.apigateway_backends[context.account_id][context.region]
+
+
+def remove_empty_attributes_from_rest_api(rest_api: RestApi, remove_tags=True):
+    if not rest_api.get("binaryMediaTypes"):
+        rest_api.pop("binaryMediaTypes", None)
+    if remove_tags:
+        if not rest_api.get("tags"):
+            rest_api.pop("tags", None)
+    else:
+        rest_api["tags"] = {}
```

Comment:
Right yes, this was wrong! Thanks a lot for picking this up.
I'm not sure what tags could be, maybe they could be `None` but we would like them to be an empty dict. From the model, they're supposed to be an empty dict if not set, but I'd rather be sure with that.

I've replaced it with: 
```python
if not rest_api.get("tags"):
    if remove_tags:
        rest_api.pop("tags", None)
    else:
       # if `tags` is falsy, set it to an empty dict
        rest_api["tags"] = {}
```
I guess now the behaviour is in line, and it's clear? Thanks again for picking this up, this would have been a weird bug to see 😅